xcm/xcm-executor/src/lib.rs

```diff
@@ -271,13 +279,24 @@ impl<Config: config::Config> XcmExecutor<Config> {
 			appendix_weight: 0,
 			transact_status: Default::default(),
 			fees_mode: FeesMode { jit_withdraw: false },
+			topic: None,
 			_config: PhantomData,
 		}
 	}
 
 	/// Execute the XCM program fragment and report back the error and which instruction caused it,
 	/// or `Ok` if there was no error.
 	pub fn execute(&mut self, xcm: Xcm<Config::Call>) -> Result<(), ExecutorError> {
+		let message_hash =
+			VersionedXcm::from(xcm.clone()).using_encoded(sp_io::hashing::blake2_256);
```

Comment:
This won't really work since the hash is not of the message which was used for transport. Also the message hash will never be reported back to the caller.

What is the use-case of this function? If just tests, then it should be marked as such.